File: blazerod/render/src/main/java/top/fifthlight/blazerod/extension/RenderPipelineBuilderExt.java
package top.fifthlight.blazerod.extension;

import com.mojang.blaze3d.vertex.VertexFormat;
import org.jetbrains.annotations.NotNull;

public interface RenderPipelineBuilderExt {
    void blazerod$withVertexFormat(@NotNull VertexFormat format);

    void blazerod$withStorageBuffer(@NotNull String name);
}

File: blazerod/render/src/main/kotlin/top/fifthlight/blazerod/extension/RenderPipelineBuilderExt.kt
package top.fifthlight.blazerod.extension

import com.mojang.blaze3d.pipeline.RenderPipeline
import com.mojang.blaze3d.vertex.VertexFormat

fun RenderPipeline.Builder.withVertexFormat(format: VertexFormat) = also {
    (this as RenderPipelineBuilderExt).`blazerod$withVertexFormat`(format)
}

fun RenderPipeline.Builder.withStorageBuffer(name: String) = also {
    (this as RenderPipelineBuilderExt).`blazerod$withStorageBuffer`(name)
}
